kotlintest-assertions/src/main/kotlin/io/kotlintest/Failures.kt

```diff
@@ -1,5 +1,23 @@
 package io.kotlintest
 
+/**
+ * Verifies that [block] throws an [AssertionError]
+ *
+ * If [block] throws an [AssertionError], this method will pass. Otherwise, it will throw an error, as a failure was
+ * expected
+ */
+fun shouldFail(block: () -> Any?) {
+  val passed = try {
+    block()
+    true
+  } catch (_: Throwable) {
```

Comment:
This will catch any exception or error. 
I think if it throws something that is not AssertionError we should just let the error propagate? Otherwise the test case will not have the correct exception attached to it.

Reply:
I was thinking that maybe users that use this method wouldn't care about anything, just that this code fails. Do you think it's better with just AssertionError?

Reply:
I think it's better with just AssertionError.

You have two possibilities.

1. I want to check that my code throws an exception - shouldThrow and it's variants
2. I want to check that my code fails an assertion - shouldFail

And the downside to catching everything with 2 is that if your test fails for some other reason - FileNotFoundException say - you won't see _that_ exception, but _our_ error message. Exceptions are properly propagated out to the junit platform as it stands - they have different statues for "failed" and "errored".

Reply:
Makes sense to me. I'll change shouldFail to AssertionError again. Good one!

Reply:
Fixed.